File: src/main/java/dev/elysium/eraces/bootstrap/CommandInitializer.java
package dev.elysium.eraces.bootstrap;

import dev.elysium.eraces.ERaces;
import dev.elysium.eraces.commands.AbilsCommand;
import dev.elysium.eraces.commands.MyraceCommand;
import dev.elysium.eraces.commands.RacesCommand;
import dev.elysium.eraces.exceptions.internal.InitFailedException;
import io.papermc.paper.plugin.lifecycle.event.types.LifecycleEvents;

public class CommandInitializer implements IInitializer {
    @Override
    public void setup(ERaces plugin) {
        try {
            plugin.getLifecycleManager().registerEventHandler(LifecycleEvents.COMMANDS, commands -> {
                commands.registrar().register(new MyraceCommand().getCmd());
                commands.registrar().register(new RacesCommand().getCmd());
                commands.registrar().register(new AbilsCommand().getCmd());
            });
        } catch (Exception e) {
            throw new InitFailedException("Ошибка при регистрации команд", e);
        }
    }
}

File: src/main/java/dev/elysium/eraces/bootstrap/CommandInitializer.kt
package dev.elysium.eraces.bootstrap

import dev.elysium.eraces.ERaces
import dev.elysium.eraces.commands.AbilsCommand
import dev.elysium.eraces.commands.MyRaceCommand
import dev.elysium.eraces.commands.RacesCommand
import dev.elysium.eraces.exceptions.internal.InitFailedException
import io.papermc.paper.command.brigadier.Commands
import io.papermc.paper.plugin.lifecycle.event.handler.LifecycleEventHandler
import io.papermc.paper.plugin.lifecycle.event.registrar.ReloadableRegistrarEvent
import io.papermc.paper.plugin.lifecycle.event.types.LifecycleEvents

class CommandInitializer : IInitializer {
    override fun setup(plugin: ERaces) {
        try {
            plugin.lifecycleManager.registerEventHandler(
                LifecycleEvents.COMMANDS,
                LifecycleEventHandler { commands: ReloadableRegistrarEvent<Commands> ->
                    listOf(MyRaceCommand.cmd, RacesCommand.cmd, AbilsCommand.cmd)
                        .forEach { commands.registrar().register(it) }
                }
            )
        } catch (e: Exception) {
            throw InitFailedException("Ошибка при регистрации команд", e)
        }
    }
}
